plugins/modules/inventory_intent.py

```diff
@@ -0,0 +1,689 @@
+        """
+
+        device_added = False
+        device_updated = False
+        # device_resynced = False
+
+        devices_to_add = self.have["device_not_in_dnac"]
+        self.result['msg'] = []
+
+        if not devices_to_add:
+            # Write code for device updation
+            device_updated = True
+
+            log("Devices {0} are present in DNAC and updated successfully".format(config['ipAddress']))
+            msg = "Devices {0} present in DNAC and updated successfully".format(config['ipAddress'])
+            self.result['msg'].append(msg)
+            self.status = "success"
+            return self
+
+        else:
+            # If we want to add device in inventory
+            config = self.mandatory_parameter(config)
+            config['ipAddress'] = devices_to_add
+            del config['resync']
+
+            try:
+                response = self.dnac._exec(
+                    family="devices",
+                    function='add_device',
+                    op_modifies=True,
+                    params=config,
+                )
+
+                log(str(response))
+                device_added = True
+
+            except Exception as e:
+                log("There is error while adding devices in DNAC")
+
+        if device_added or device_updated:
+            if response and isinstance(response, dict):
+                task_id = response.get('response').get('taskId')
+                while True:
+                    execution_details = self.get_task_details(task_id)
+
+                    if '/task/' in execution_details.get("progress"):
+                        self.result['changed'] = True
+                        self.result['response'] = execution_details
+                        break
+
+                    elif execution_details.get("isError") and execution_details.get("failureReason"):
+                        self.module.fail_json(msg=execution_details.get("failureReason"),
+                                              response=execution_details)
+                        break
+
+                log("Device Added Successfully")
+                log("Added devices are :" + str(devices_to_add))
+                msg = "Device " + str(devices_to_add) + " added Successfully !!"
+                self.result['msg'].append(msg)
+
+        return self
+
+    def get_diff_deleted(self, config):
+        """
+        Delete devices in DNAC based on provided inputs.
+        Args:
+            config (dict): A dictionary containing the list of device IP addresses to be deleted.
+        Returns:
+            object: An instance of the class with updated results and status based on the deletion operation.
+        Description:
+            This function is responsible for removing devices from the DNAC inventory.
+        """
+
+        device_to_delete = config.get("ipAddress")
+        self.result['msg'] = []
+
+        for device_ip in device_to_delete:
```

Comment:
can we change the flow like below for better clarity? if device_ip is **not** there then continue 
```
   for device_ip in config.get("ipAddress"):
        if device_ip not in self.have.get("device_in_dnac"):
            self.result['changed'] = False
            msg = f"The device {device_ip} is not present in Cisc DNA Center, so can't perform the delete operation."
            self.result['msg'].append(msg)
            continue

        device_id = self.get_device_id(device_ip)

```
So you don't need to have one else case.. (the last else case) and makes the flow good